services-js/commissions-app/src/server/services/Email.ts

```diff
@@ -0,0 +1,117 @@
+import { Client as PostmarkClient } from 'postmark';
+import Rollbar from 'rollbar';
+
+import { Commission } from '../graphql/schema';
+
+import EmailContent, {
+  TemplateData as ApplicationData,
+} from '../email/EmailContent';
+
+// see services-js/registry-certs/server/services/Emails.ts
+
+/**
+ * Sends a confirmation email to the applicant, as well as to the policy office.
+ */
+export default class Email {
+  postmarkClient: PostmarkClient;
+  rollbar: Rollbar;
+  emailContent: EmailContent;
+  // todo: need commissions array
+  commissions: Commission[];
+
+  applicantSubject = 'We’ve received your application';
+  policyOfficeSubject = 'New City Clerk Boards and Commissions Application';
+  policyOfficeToAddress = process.env.POLICY_OFFICE_TO_ADDRESS ||
+  'boardsandcommissions@boston.gov';
+
+  constructor(postmarkClient: PostmarkClient, rollbar: Rollbar) {
+    this.postmarkClient = postmarkClient;
+    this.rollbar = rollbar;
+
+    this.emailContent = new EmailContent();
+
+    // todo: need commissions array
+    this.commissions = [];
+  }
+
+  // todo: need newId from db server
+  sendConfirmations(email, newId, commissionIds) {
+    const applicationData: ApplicationData = {
+      // todo: need commissions array
+      commissions: namesFromIds(this.commissions, commissionIds),
+      email,
+      newId,
+    };
+
+    const renderedApplicantBodies = this.emailContent.renderApplicantBodies(
+      applicationData
+    );
+    const renderedPolicyOfficeBodies = this.emailContent.renderPolicyOfficeBodies(
+      applicationData
+    );
+
+    // send to applicant...
+    this.sendConfirmationEmail(
+      applicationData.email,
+      this.applicantSubject,
+      renderedApplicantBodies
+    ).catch(err => this.rollbar.error(`Error sending applicant email: ${err}`));
+
+    // send to policy office...
+    this.sendConfirmationEmail(
+      this.policyOfficeToAddress,
+      this.policyOfficeSubject,
+      renderedPolicyOfficeBodies
+    ).catch(err =>
+      this.rollbar.error(`Error sending policy office email: ${err}`)
+    );
+  }
+
+  /**
+   * Use Postmark to send an email. The user does not need to be notified
+   * if there is an error.
+   */
+  async sendConfirmationEmail(
+    emailTo: string,
+    subject: string,
+    renderedBodies
+  ): Promise<void> {
+    try {
+      await new Promise((resolve, reject) => {
+        PostmarkClient.sendEmail(
+          {
+            To: emailTo,
+            From: 'no-reply@boston.gov',
+            Subject: subject,
+            HtmlBody: renderedBodies.html,
+            TextBody: renderedBodies.text,
+          },
+          (err, result) => (err ? reject(err) : resolve(result))
+        );
+      });
+    } catch (err) {
+      // todo: ? confirm this will work as expected
+      return Promise.reject(err);
```

Comment:
You don’t need this bit. You actually don’t need to make the function `async`. If an exception is thrown in the `Promise` constructor function, that automatically just rejects the promise. So you can `return new Promise(…)` and that does the right thing.